script.js

```diff
@@ -1,4 +1,400 @@
+let noOfPlayers = 0;
+let playerProfiles = [];
+let dealerCards = [];
+let deck = [];
+let activePlayer = 0;
+let gameMode = 'start';
+let roundsPlayed = 0;
+
+const inputEl = document.querySelector('#input-field');
+const newGameBtn = document.querySelector('.new-game');
+const outputEl = document.querySelector('#output-div');
+const scoreboardEl = document.querySelector('#scoreboard-data');
+
+/* TODO
+add DOM elements for:
+1. New game button (done)
+2. Track wins of each player (done)
+3. Reshuffle deck when current deck is empty
+*/
+
+// DOM --->
+// press 'enter' to submit
+inputEl.addEventListener('keydown', function (e) {
+  if (e.key === 'Enter') {
+    document.querySelector('#submit-button').click();
+  }
+  newGameBtn.addEventListener('click', function () {
+    newGame();
+  });
+});
+
+// <--- DOM
+
+/*
+dynamic scoretable to track each player's win/loss
+1. scoretable is created when amt of players are input
+2. scoretable is updated at the end of each round
+
+scoretable to contain:
+<player name> = win: <>, draw: <>, loss: <>
+**can use spread operator
+*/
+
+/////////////////
+/// MAIN FUNCTION
+////////
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameMode === 'start') {
+    inputEl.placeholder = 'Follow instructions in the output.';
+    // logic for 1st round of every game
+    if (roundsPlayed === 0) {
+      if (input >= 1) {
+        noOfPlayers = input;
+        newUserProfile(noOfPlayers);
+        dealCards(noOfPlayers);
+        // next game mode
+        gameMode = 'blackJackCheck';
+        return `Cards have been dealt to the ${noOfPlayers} players.`;
+      } else if (!noOfPlayers) {
+        return `Please enter the number of player(s) playing. (at least 1)`;
+      }
+      // logic for every round after 1st round
+    } else {
+      activePlayer = 0;
+      resetCards();
+      dealCards(noOfPlayers);
+      // next game mode
+      gameMode = 'blackJackCheck';
+      inputEl.placeholder = 'Click submit.';
+      return `Cards have been dealt to the ${noOfPlayers} players again.`;
+    }
+  }
+  if (gameMode === 'blackJackCheck') {
+    scoreboardEl.textContent = outputScoreboardData();
+    // check if cards are blackjack for active player
+    // if player's cards are blackjack, output if it is.
+    if (activePlayer < playerProfiles.length) {
+      if (blackJackCheck(playerProfiles[activePlayer].hand)) {
+        playerProfiles[activePlayer].blackjack = 1;
+        let output = `Player ${activePlayer + 1} has blackjack!`;
+        activePlayer += 1;
+        return output;
+      }
+      gameMode = 'hit';
+      // FIXME can i remove this return??
```

Comment:
Solely looking at the logic, I believe this `return` is still required. Otherwise, the game mode will move immediately in the `'hit'` game mode without any feedback indicating so to the user(s).

Besides, notice on [lines 96 to 98](https://github.com/rocketacademy/basics-blackjack/blob/018304d92e7fa46f1cb9a5dc03263896c3d6f6db/script.js#L96-L98), you are using an `else if` instead of `if`. If you are using `if`, setting `gameMode` to `'hit'` [here](https://github.com/rocketacademy/basics-blackjack/blob/018304d92e7fa46f1cb9a5dc03263896c3d6f6db/script.js#L84) will cause the condition [here](https://github.com/rocketacademy/basics-blackjack/blob/018304d92e7fa46f1cb9a5dc03263896c3d6f6db/script.js#L96) to be met and thus, enter [this block](https://github.com/rocketacademy/basics-blackjack/blob/018304d92e7fa46f1cb9a5dc03263896c3d6f6db/script.js#L96-L98).

However, with an `else if`, it will simply skip over it, because it assumes that this `else if` is chained to this [if block](https://github.com/rocketacademy/basics-blackjack/blob/018304d92e7fa46f1cb9a5dc03263896c3d6f6db/script.js#L73). This "blackJackCheck" game mode condition is originally met, and the next conditional is an `else if` on [line 96](https://github.com/rocketacademy/basics-blackjack/blob/018304d92e7fa46f1cb9a5dc03263896c3d6f6db/script.js#L96), so it assumes the next conditional is not met despite the fact that you changed the `gameMode` to 'hit' in [line 84](https://github.com/rocketacademy/basics-blackjack/blob/018304d92e7fa46f1cb9a5dc03263896c3d6f6db/script.js#L84).

All conditions thereafter are also not met, so it skips over to [line 179](https://github.com/rocketacademy/basics-blackjack/blob/018304d92e7fa46f1cb9a5dc03263896c3d6f6db/script.js#L179) to return "Starting next round!" Do you intend to show this message when the player does not draw a Blackjack on the "blackJackCheck" `gameMode`?